src/commands/action/api_vip_connectivity_check.go

```diff
@@ -0,0 +1,30 @@
+package action
+
+import (
+	"github.com/openshift/assisted-installer-agent/src/config"
+	"github.com/openshift/assisted-service/models"
+)
+
+type apiVipConnectivityCheck struct {
+	args []string
+}
+
+func (a apiVipConnectivityCheck) Validate() error {
+	err := validateCommon("api vip connectivity check", 1, a.args, a.args[0], models.APIVipConnectivityRequest{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (a apiVipConnectivityCheck) Run() (string, []string) {
+	podmanRunCmd := []string{
+		"run", "--privileged", "--net=host", "--rm", "--quiet",
+		"-v", "/var/log:/var/log",
+		"-v", "/run/systemd/journal/socket:/run/systemd/journal/socket",
+		config.GlobalAgentConfig.AgentVersion,
+		"apivip_check",
+	}
+
+	return podman, append(podmanRunCmd, a.args...)
```

Comment:
if you create an instance of models.APIVipConnectivityRequest from the args, here you could do:
```suggestion
	return podman, append(podmanRunCmd, apiVipConnectivityRequest.URL)
```
Which is less generic but clearer.

Reply:
we can't do it, as the currently it expects dump json and will be unmarshalled in api_vip command 
Till we don't change the way all our cmds works we will need to dump each time and there is no point for it as args have needed value